components/provisioner/internal/gardener/provisioner.go

```diff
@@ -111,8 +113,16 @@ func (g *GardenerProvisioner) UpgradeCluster(clusterID string, upgradeConfig mod
 		return appErr.Append("error while updating Gardener shoot configuration")
 	}
 
+	setObjectFields(shoot)
```

Comment:
Do we need to set `kind` and `APIVersion`? Shouldn't the `shoot` variable already contain those (the `Get` method is returning result from `github.com/gardener/gardener/pkg/apis/core/v1beta1/types` package).

Reply:
Unfortunately, `Get()` returns empty fields `kind` and `APIVersion`. I discovered this during a debugging session of the first version of the code, which had no manualy assignment of `kind` and `APIVersion` values. The other fields of the `Shoot` object are filled correctly.